runtime/compiler/runtime/IProfiler.cpp

```diff
@@ -3153,7 +3153,7 @@ TR_IPBCDataCallGraph::loadFromPersistentCopy(TR_IPBCDataStorageHeader * storage,
          if (comp->fej9()->sharedCache()->isROMClassOffsetInSharedCache(csInfoClazzOffset, &romClass))
             ramClass = ((TR_J9VM *)comp->fej9())->matchRAMclassFromROMclass((J9ROMClass *)romClass, comp);
 
-         if (ramClass)
+         if (ramClass && comp->fej9()->isClassInitialized((TR_OpaqueClassBlock*)ramClass))
```

Comment:
Might be worth adding a comment here about why this is necessary (basically what it results in the optimizer) and also why we're ok with having the ith entry be NULL (essentially just a brief summary of Marius' comments).